### PR TITLE
[#4472 ] fix(CI pipeline):  Fixed Doris test failure as a result of excessive disk usage

### PR DESCRIPTION
**Pull request by yuqi1129**
### What changes were proposed in this pull request?

Remove unnecessary files such as Android and Haskell.

### Why are the changes needed?

To free more space to reduce test failure rate.

image

Fix: #4472 

### Does this PR introduce _any_ user-facing change?

N/A

### How was this patch tested?

Locally and CI.

### CODE REVIEW

File: dev/ci/util_free_space.sh [hunk 1]
```diff
@@ -49,6 +49,10 @@ if [ "${GITHUB_ACTIONS}" = "true" ]; then
   sudo rm -rf /opt/hostedtoolcache/PyPy || :
   # 376MB
   sudo rm -rf /opt/hostedtoolcache/node || :
+  # free at least 10G, android is unless for Gravitino CI
```
Comment (jerryshao) [inline dev/ci/util_free_space.sh:52]: “useless”, not "unless".

**Review by jerryshao — APPROVED** [review]

File: dev/ci/util_free_space.sh (hunk 1 above)
Comment (zhoukangcn) [inline dev/ci/util_free_space.sh:52]: just a typo `unless`, others LGTM

**Review by zhoukangcn — DISMISSED** [review]

File: dev/ci/util_free_space.sh (hunk 1 above)
Comment (yuqi1129) [reply dev/ci/util_free_space.sh]: Fixed.

**Review by jerryshao — APPROVED** [review]

**Review by zhoukangcn — APPROVED** [review]

**Comment by jerryshao** [comment]
I will merge this when CI is passed, no need to test again. @yuqi1129

**Comment by yuqi1129** [comment]
> I will merge this when CI is passed, no need to test again. @yuqi1129

Got it.

**Pull request merged**